src/test/java/com/google/jenkins/plugins/k8sengine/CredentialsUtilTest.java

```diff
@@ -0,0 +1,131 @@
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CredentialsUtilTest {
+  private static final String TEST_CREDENTIALS_ID = "test-credentials-id";
+  private static final String TEST_INVALID_CREDENTIALS_ID = "test-invalid-credentials-id";
+
+  @ClassRule public static JenkinsRule r = new JenkinsRule();
+
+  public static Jenkins jenkins;
+
+  @BeforeClass
+  public static void init() throws IOException {
+    jenkins = r.jenkins;
+
+    CredentialsStore store = new SystemCredentialsProvider.ProviderImpl().getStore(jenkins);
+    GoogleRobotCredentials credentials = Mockito.mock(GoogleRobotCredentials.class);
+    credentials.getId();
+    Mockito.when(credentials.getId()).thenReturn(TEST_CREDENTIALS_ID);
+    store.addCredentials(Domain.global(), credentials);
+  }
+
+  @Test
+  public void testGetRobotCredentialsReturnsFirstCredential() throws IOException {
+    assertNotNull(
+        CredentialsUtil.getRobotCredentials(
+            jenkins.get(), ImmutableList.<DomainRequirement>of(), TEST_CREDENTIALS_ID));
+  }
+
+  @Test(expected = AbortException.class)
+  public void testGetRobotCredentialsInvalidCredentialsIdAbortException() throws AbortException {
+    CredentialsUtil.getRobotCredentials(
+        jenkins.get(), ImmutableList.<DomainRequirement>of(), TEST_INVALID_CREDENTIALS_ID);
+  }
+
+  @Test(expected = AbortException.class)
+  public void testGetGoogleCredentialAbortException()
+      throws GeneralSecurityException, AbortException {
+
+    GoogleRobotCredentials robotCreds = Mockito.mock(GoogleRobotCredentials.class);
+    Mockito.when(robotCreds.getGoogleCredential(any(ContainerScopeRequirement.class)))
+        .thenThrow(new GeneralSecurityException());
+    CredentialsUtil.getGoogleCredential(robotCreds);
+  }
+
+  @Test
+  public void testGetGoogleCredentialReturnsCredential()
+      throws GeneralSecurityException, AbortException {
+    GoogleRobotCredentials robotCreds = Mockito.mock(GoogleRobotCredentials.class);
+    Credential credential = Mockito.mock(Credential.class);
+    Mockito.when(robotCreds.getGoogleCredential(any(ContainerScopeRequirement.class)))
+        .thenReturn(credential);
+    assertNotNull(CredentialsUtil.getGoogleCredential(robotCreds));
+  }
+
+  @Test(expected = IOException.class)
+  public void testGetAccessTokenIOException() throws IOException {
+    GoogleCredential googleCredential = Mockito.mock(GoogleCredential.class);
+    Mockito.when(googleCredential.refreshToken()).thenThrow(IOException.class);
+    CredentialsUtil.getAccessToken(googleCredential);
+  }
+
+  @Test
+  public void testGetAccessTokenReturnsToken() throws IOException {
+    GoogleCredential googleCredential = Mockito.mock(GoogleCredential.class);
+    Mockito.when(googleCredential.refreshToken()).thenReturn(true);
+    Mockito.when(googleCredential.getAccessToken()).thenReturn("access token");
+    assertNotNull(CredentialsUtil.getAccessToken(googleCredential));
```

Comment:
Suggest also verifying the contents of the returned cred match expected.

Reply:
Done